src/components/modals/data-table-modal.js

```diff
@@ -30,183 +32,179 @@ import {Clock} from 'components/common/icons/index';
+      {item}
+    </div>
+  );
+});
 
-    // If canvas can not be scrolled left anymore when we try to scroll left
-    const prevent_left = e.deltaX < 0 && canvas.scrollLeft <= 0;
-    // If canvas can not be scrolled up when we try to scroll up
-    const prevent_up = e.deltaY < 0 && canvas.scrollTop <= 0;
+export class DataTableModal extends Component {
 
-    if (prevent_left || prevent_up) {
-      e.preventDefault();
-    }
-  };
+  headerGrid = React.createRef();
+
+  datasetsSelector = props => props.datasets;
+  dataIdSelector = props => props.dataId;
+  activeDatasetSelector = createSelector(
+    this.datasetsSelector,
+    this.dataIdSelector,
+    (datasets, dataId) => datasets[dataId]
+  );
+  columnsSelector = createSelector(
+    this.activeDatasetSelector,
+    dataset => dataset.fields.filter(({name}) => name !== '_geojson')
+  );
+  rowsSelector = createSelector(
+    this.activeDatasetSelector,
+    dataset => dataset.data
+  );
+
+  headerRowHeight = () => 72;
+  cellRowHeight = () => 48;
+  columnWidth = index => COLUMN_SIZE[this.columnsSelector(this.props)[index].type] || DEFAULT_COLUMN_WIDTH;
+  onRowScrolling = ({scrollLeft}) => this.headerGrid.current.scrollTo({scrollLeft});
 
   render() {
-    const {datasets, dataId, showDatasetTable} = this.props;
+    const {showDatasetTable} = this.props;
+
+    const datasets = this.datasetsSelector(this.props);
+    const dataId = this.dataIdSelector(this.props);
 
     if (!datasets || !dataId) {
       return null;
     }
 
-    const activeDataset = datasets[dataId];
-    // TODO: this should be all data
-    const rows = activeDataset.data;
-    const columns = activeDataset.fields
-      .map((field, i) => ({
-        ...field,
-        key: i,
-        headerRenderer: <FieldHeader {...field} />,
-        resizable: true,
-        formatter:
-          field.type === ALL_FIELD_TYPES.boolean ? BooleanFormatter : undefined
-      }))
-      .filter(({name}) => name !== '_geojson');
+    const activeDataset = this.activeDatasetSelector(this.props);
+    const columns = this.columnsSelector(this.props);
+    const rows = this.rowsSelector(this.props);
 
     return (
-      <StyledModal ref={this._root} className="dataset-modal" style={{overflow: 'scroll'}}>
+      <StyledModal className="dataset-modal" >
         <DatasetTabs
           activeDataset={activeDataset}
           datasets={datasets}
           showDatasetTable={showDatasetTable}
         />
-        <DataGridWrapper
-          onWheel={shouldPreventScrollBack ? this._onMouseWheel : null}
-        >
-          {ReactDataGrid && (
-            <ReactDataGrid
-              headerRowHeight={72}
-              columns={columns}
-              minColumnWidth={172}
-              minWidth={this.props.width}
-              minHeight={this.props.height - 65}
-              rowGetter={i => rows[i]}
-              rowHeight={48}
-              rowsCount={rows.length}
-            />
+        <Autosizer>
```

Comment:
is `Autosizer` really necessary here?

With Autosizer, you still see white space on the right. The table should expand to entire with of the modal

<img width="1152" alt="Screen Shot 2019-06-21 at 12 48 53 AM" src="https://user-images.githubusercontent.com/3605556/59906849-8d960800-93be-11e9-9a5f-e2ee9a3e0963.png">